Salary App/Features/SalaryForm/View/SalaryFormViewController.swift

```diff
@@ -0,0 +1,336 @@
+        stackView.layer.cornerRadius = 16
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 8
+        return stackView
+    }()
+    
+    private lazy var workingHoursLbl: UILabel = {
+        let SalaryDescdLbl = UILabel()
+        SalaryDescdLbl.textColor = .black
+        SalaryDescdLbl.text = "Enter Working hours"
+        SalaryDescdLbl.translatesAutoresizingMaskIntoConstraints = false
+        return SalaryDescdLbl
+    }()
+    
+    private lazy var workingHoursTextField: UITextField = {
+        let textField = UITextField()
+        textField.layer.borderWidth = 1
+        textField.textColor = .black
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 50))
+        textField.leftView = paddingView
+        textField.leftViewMode = .always
+        textField.keyboardType = .asciiCapableNumberPad
+        textField.layer.cornerRadius = 8
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+    
+    private lazy var taxCodeStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .equalSpacing
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layer.cornerRadius = 16
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 8
+        return stackView
+    }()
+    
+    private lazy var taxCodeLbl: UILabel = {
+        let taxCodeLbl = UILabel()
+        taxCodeLbl.textColor = .black
+        taxCodeLbl.text = "Enter Tax code: 1257L is the most common tax code"
+        taxCodeLbl.numberOfLines = 0
+        taxCodeLbl.translatesAutoresizingMaskIntoConstraints = false
+        return taxCodeLbl
+    }()
+    
+    private lazy var taxCodeTextField: UITextField = {
+        let textField = UITextField()
+        textField.layer.borderWidth = 1
+        textField.placeholder = "1257L"
+        textField.text = "1257L"
+        textField.textColor = .black
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 50))
+        textField.leftView = paddingView
+        textField.leftViewMode = .always
+        textField.keyboardType = .asciiCapableNumberPad
+        textField.layer.cornerRadius = 8
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+    
+    private lazy var calculateBtn: UIButton = {
+        let calculateBtn = UIButton()
+        calculateBtn.translatesAutoresizingMaskIntoConstraints = false
+        calculateBtn.backgroundColor = .systemBlue
+        calculateBtn.setTitle("Submit", for: .normal)
+        calculateBtn.layer.cornerRadius = 16
+        calculateBtn.addTarget(self, action: #selector(calculateSalary), for: .touchUpInside)
+        return calculateBtn
+    }()
+    
+    
+   private let validationManager = ValidationManager()
```

Comment:
You're creating the same objects twice validationManager, taxCalculator, niCalculator, studentLoanCalulator. All of these also created inline.

If you want to init the salaryVm constant why don't you init everything in an init like

```


private let validationManager = ValidationManager()
   private let taxCalculator = TaxCalculator()
   private let niCalculator = NICalaculator()
   private let pensionCalculator = PensionContributionCalculator()

   private let salaryvm:  SalaryFormViewModel!

   init() {
   self.salaryVm = SalaryFormViewModel(taxCalculator: taxCalculator, niCalculator: niCalculator, studentLoanCalculator: studentLoanCalulator, pensionContributionCalculator: pensionCalculator)
    super.init()...
   
  
   }



```